packages/synapse-react-client/src/components/download_list/TableQueryDownloadConfirmation.tsx

```diff
@@ -9,17 +9,36 @@ import { DownloadConfirmationUI } from './DownloadConfirmationUI'
 import { useQueryContext } from '../QueryContext'
 import { useQueryVisualizationContext } from '../QueryVisualizationWrapper'
 import { displayFilesWereAddedToDownloadListSuccess } from './DownloadConfirmationUtils'
+import { getValueFilter } from '../../utils/functions/QueryFilterUtils'
 
 export function TableQueryDownloadConfirmation() {
-  const { getLastQueryRequest } = useQueryContext()
-  const { setShowDownloadConfirmation } = useQueryVisualizationContext()
+  const { data, getLastQueryRequest } = useQueryContext()
+  const {
+    setShowDownloadConfirmation,
+    hasSelectedRows,
+    selectedRows,
+    rowSelectionPrimaryKey,
+  } = useQueryVisualizationContext()
   const queryBundleRequest = useMemo(() => {
     const requestCopy = getLastQueryRequest()
     requestCopy.partMask =
       SynapseConstants.BUNDLE_MASK_QUERY_COUNT |
       SynapseConstants.BUNDLE_MASK_SUM_FILES_SIZE_BYTES
+
+    if (hasSelectedRows && rowSelectionPrimaryKey && data?.columnModels) {
+      requestCopy.query.additionalFilters = [
+        ...(requestCopy.query.additionalFilters || []),
+        getValueFilter(rowSelectionPrimaryKey, selectedRows, data.columnModels),
+      ]
+    }
```

Comment:
This gives us the functionality. If rows are selected, apply the filter to the download list query.